langserver/config.go

```diff
@@ -36,27 +46,25 @@ func (s *server) DidChangeConfiguration(ctx context.Context, params *protocol.Di
 			Message: fmt.Sprintf("Received notification change: %v\n", params),
 		})
 
-	setting := params.Settings
+	config := localLSPConfiguration{}
 
-	// the struct expected is the following
-	// promql:
-	//   url: http://
-	//   interval: 3w
-	m, ok := setting.(map[string]map[string]string)
-	if !ok {
+	// In order to avoid a loop other different interface coming from the general settings
+	// and in order to avoid a systematic cast at each steps, we are going to marshall and unmarshall again the configuration
+	// with the expected struct (i.e. localLSPConfiguration).
+	rawData, marshallError := json.Marshal(params.Settings)
+	if marshallError != nil {
 		// nolint: errcheck
 		s.client.LogMessage(ctx, &protocol.LogMessageParams{
 			Type:    protocol.Error,
-			Message: fmt.Sprint("unexpected format of the configuration"),
+			Message: fmt.Sprint("unable to serialize the configuration"),
 		})
 		return nil
 	}
-	config, ok := m["promql"]
-	if !ok {
+	if err := json.Unmarshal(rawData, &config); err != nil {
 		// nolint: errcheck
 		s.client.LogMessage(ctx, &protocol.LogMessageParams{
 			Type:    protocol.Error,
-			Message: fmt.Sprint("promQL key not found"),
+			Message: fmt.Sprint("unexpected format of the configuration"),
```

Comment:
```suggestion
			Message: fmt.Sprint("unexpected configuration format"),
```

Can you include the error message from the json package here, e.g. via `Wrapf`?

Reply:
I don't see the point of using `Wrapf` since the error is not used afterward and also since the `Message` is taking a string and not an error.

We could just do that instead : 


```go
Message: fmt.Sprintf("unexpected configuration format: %s",err),

Reply:
That'd be equivalent to 

```
Message: errors.Wrap(err, "unexpected configuration format").Error(),
```

I like this version slightly better, but there isn't a significant difference.